RELEASES.md

```diff
@@ -1,3 +1,82 @@
+Version 1.31.0 (2018-12-06)
+==========================
+
+Language
+--------
+- [This version marks the release of the 2018 edition of Rust.][54057]
+- [There is now an explicit attribute for declaring a trait is a
+  _"marker"_ trait.][53693] Using `#[marker]` will declare that the trait is a
+  marker trait which requires that the trait doesn't have any items and that
+  impls must be empty. This allows you to have overlapping implementations.
+- [New lifetime elision rules now allow for eliding lifetimes in functions and
+  impl headers.][54778] E.g. `impl<'a> Reader for BufReader<'a> {}` can now be
+  `impl Reader for BufReader {}`. Lifetimes are still required to be defined
+  in structs.
+- [You can now define and use constant functions.][54835] These are currently
+  a strict minimal subset of the [const fn RFC][RFC-911]. Refer to the [tracking
+  issue][53555] for what exactly is available.
+- [You can now use tool lints, which allow you to scope lints from external
+  tools using attributes.][54870] E.g. `#[allow(clippy::filter_map)]`.
+- [`#[no_mangle]` and `#[export_name]` attributes can now be located anywhere in
+  a crate, not just in exported functions.][54451]
+- [You can now use parentheses in pattern matches.][54497]
+
+Compiler
+--------
+- [Updated musl to 1.1.20][54430]
+
+Libraries
+---------
+- [You can now convert `num::NonZero*` types to their raw equivalvents using the
+  `From` trait.][54240] E.g. `u8` now implements `From<NonZeroU8>`.
+- [You can now convert a `&Option<T>` into `Option<&T>` and `&mut Option<T>`
+  into `Option<&mut T>` using the `From` trait.][53218]
+- [You can now multiply (`*`) a `time::Duration` by a `u32`.][52813]
+- [`mem::needs_drop` is now a const fn.][54596]
```

Comment:
Not on stable:
```rust
#[rustc_const_unstable(feature = "const_needs_drop")]
```
https://doc.rust-lang.org/nightly/src/core/mem.rs.html#461-463